src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/PeriodicSyncStrategy.java

```diff
@@ -28,25 +77,30 @@ public PeriodicSyncStrategy(Retryer retryer) {
      */
     @Override
     public void start(SyncContext context, int syncParallelism) {
-
+        super.startSync(context, syncParallelism);
     }
 
     @Override
     void doStart(SyncContext context, int syncParallelism) {
-
+        logger.atInfo(SYNC_EVENT_TYPE).kv("interval", interval).log("Start periodic syncing");
+        this.scheduledFuture = syncExecutorService
+                .scheduleAtFixedRate(this::syncLoop, 0, interval, TimeUnit.SECONDS);
     }
 
     @Override
     void doStop() {
-
+        logger.atDebug(SYNC_EVENT_TYPE).log("Cancel {} periodic sync thread(s)", syncThreads.size());
+        if (this.scheduledFuture != null && !this.scheduledFuture.isCancelled()) {
+            this.scheduledFuture.cancel(true);
+        }
     }
 
     /**
      * Stops the syncing of shadows.
      */
     @Override
     public void stop() {
-
+        super.stopSync();
```

Comment:
If you made the `BaseSyncStrategy` implement `SyncStrategy` then could `start(context, syncParallelism)`, `stop()` and `putSyncRequest(SyncRequest request)` go in the base class?

Reply:
They are ever so slightly different. I have put the common functionality in the base class as much as possible.

Reply:
what's different?
putSyncRequest looks to be same implementation in both Periodic and Realtime
start is the same
stop is the same

Reply:
There's no difference here:

`start` in `PeriodicSyncStrategy` (line 81) is:
```
    public void start(SyncContext context, int syncParallelism) {
        super.startSync(context, syncParallelism);
    }

```

`start` in `RealTimeSyncStrategy` (line 70) is :
```
@Override
    public void start(SyncContext context, int syncParallelism) {
        super.startSync(context, syncParallelism);
    }
 ```